meeteval/wer/wer/time_constrained.py

```diff
@@ -370,12 +381,53 @@ def sort_segments(s: TimeMarkedTranscript):
     )
 
 
+def sort_and_validate( segments, sort, pseudo_word_level_timing, name):
+    # Check that all timings are valid
+    if len(segments.transcript) != len(segments.timings):
+        raise ValueError(
+            f'Number of words does not match number of timings in {name}: '
+            f'{len(segments.transcript)} != {len(segments.timings)}'
+        )
+
+    for t in segments.timings:
+        if t[1] < t[0]:
+            raise ValueError(f'The end time of an interval must be larger than the start time. Found {t} in {name}')
+
+    prune = True
+
+    assert sort in (True, False, 'segment', 'word')
+    if sort in (True, 'segment', 'word'):
+        segments = sort_segments(segments)
+
+    words = get_pseudo_word_level_timings(segments, pseudo_word_level_timing)
+
+    # Check whether words are sorted by start time
+    words_sorted = sort_segments(words)
+    if words_sorted != words:
+        if sort is not True:
+            logger.warning(f'The order of word-level timings contradicts the segment-level order in {name}')
```

Comment:
How about changing the warning to
```
contradictions = [a!=b for a, b in zip(words_sorted.transcript, words.transcript)]
f'The order of word-level timings contradicts the segment-level order in {name}: {sum(contradictions)} of {len(contradictions)} times.'
```
Assuming the contradictions are local, this will give a hint, how often that happens.

Reply:
They are not guaranteed to be local. In case of a back-channel scenario this number could become large although only a single word is in the wrong position

Reply:
With local I mean, that even in those cases the number should be small. e.g. LibriCSS eval has 11159 Words, and since it should rarely happen, that something went wrong, a report of `100 of 11159` vs `5000 of 11159` would help the user, that in the latter case something went wrong.

Reply:
Okay, I tested your suggestion on the Chime7 baseline data and the numbers seem reasonable.